clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -588,17 +588,19 @@ static void InitializeStandardPredefinedMacros(const TargetInfo &TI,
   if (LangOpts.HIP) {
     Builder.defineMacro("__HIP__");
     Builder.defineMacro("__HIPCC__");
-    Builder.defineMacro("__HIP_MEMORY_SCOPE_SINGLETHREAD", "1");
-    Builder.defineMacro("__HIP_MEMORY_SCOPE_WAVEFRONT", "2");
-    Builder.defineMacro("__HIP_MEMORY_SCOPE_WORKGROUP", "3");
-    Builder.defineMacro("__HIP_MEMORY_SCOPE_AGENT", "4");
-    Builder.defineMacro("__HIP_MEMORY_SCOPE_SYSTEM", "5");
     if (LangOpts.CUDAIsDevice)
       Builder.defineMacro("__HIP_DEVICE_COMPILE__");
     if (LangOpts.GPUDefaultStream ==
         LangOptions::GPUDefaultStreamKind::PerThread)
       Builder.defineMacro("HIP_API_PER_THREAD_DEFAULT_STREAM");
   }
+  if (LangOpts.HIP || (LangOpts.OpenCL && TI.getTriple().isAMDGPU())) {
```

Comment:
Please add a test for this functionality.

Reply:
And for `SYCL_USE_NATIVE_FP_ATOMICS`

Reply:
Tests added